src/badger/gui/acr/components/routine_page.py

```
@@ -962,6 +962,25 @@ def _compose_routine(self) -> Routine:
             # Patch the BO generators to turn on TuRBO by default
             if "turbo_controller" not in generator_params:
                 generator_params["turbo_controller"] = "optimize"
+
+            # Nullify a few properties in turbo that can cause issues
+            turbo_config = generator_params["turbo_controller"]
+            if type(turbo_config) is dict:
+                if turbo_config["name"] == "optimize":
+                    try:
```

Please remove the try except statements here and below, if resetting the turbo config fails, we need it to do so in a verbose fashion

---

Those are for the cases where there are no such fields, such as when the user selects and configs the BO generator manually (sometimes they only config the fields they care about), in those cases the turbo_controller read from the generator param editor would be a dict but may not contain the `center_x` and `best_value` fields. We only silently pass on those cases (`expect KeyError`)

---

Good point, but I don't think KeyError will actually be raised here if the key doesn't already exist. It would just add the field to the dict

---

Ah right, nice catch, thanks! Will get the value first then!

---

Oh wait, actually we can safely set the fields to `None` I guess as long as turbo_controller is a dict

---

But would it matter if originally there are no such fields in the `turbo_controller` but we forcefully add in those fields and set them to `None`?